DAL/Repositories/UserRepository.cs

```diff
@@ -43,16 +43,22 @@ public async Task CreateAsync(User entity)
 
         public IEnumerable<User?> ReadByCondition(Expression<Func<User, bool>> predicate)
         {
-            return _context.Set<User>().Where(predicate).ToList();
+            return _context.Set<User>().Where(predicate).AsEnumerable();
```

Comment:
Is it good idea to not send query to database?
https://stackoverflow.com/questions/17968469/whats-the-differences-between-tolist-asenumerable-asqueryable
I think is better to keep ToList, because we will not add more filters to it.

Reply:
Yes, indeed, it would be better to use ToList in our case